src/Etl/EtlProvider.php

```diff
@@ -0,0 +1,130 @@
+    {
+        try {
+            $response = $this->client->post("/api/etl/$id", [
+                "headers" => [
+                    "Content-Type" => "application/json",
+                ],
+            ]);
+
+            switch ($response->getStatusCode()) {
+                case 200:
+                    return SerializerFactory::getInstance()->deserialize(
+                        $response->getBody()->getContents(),
+                        ExtractTransformLoadResponseDto::class,
+                        'json'
+                    );
+                case 403:
+                    throw $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_FORBIDDEN);
+                case 404:
+                    throw $this->errorFactory->create(EtlErr::ETL_CONFIG_NOT_FOUND);
+                default:
+                    throw $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_LOAD_FAILED);
+            }
+        } catch (Exception $err) {
+            throw ErrorHelper::getSherlError($err, $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_LOAD_FAILED));
+        }
+    }
+    /**
+     * Executes an Extract Transform Load process with a given configuration.
+     *
+     * @param ExtractTransformLoadInputDto $config - The configuration for the ETL process.
+     * @return ExtractTransformLoadResponseDto|null - The response from the ETL process.
+     * @throws SherlException
+     */
+    public function extractTransformLoad(ExtractTransformLoadInputDto $config): ?ExtractTransformLoadResponseDto
+    {
+        try {
+            $response = $this->client->post("/api/etl", [
+                "headers" => [
+                    "Content-Type" => "application/json",
+                ],
+                RequestOptions::JSON => $config,
+            ]);
+
+            switch ($response->getStatusCode()) {
+                case 200:
+                    return SerializerFactory::getInstance()->deserialize(
+                        $response->getBody()->getContents(),
+                        ExtractTransformLoadResponseDto::class,
+                        'json'
+                    );
+                case 403:
+                    throw $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_LOAD_FORBIDDEN);
+                default:
+                    throw $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_LOAD_FAILED);
+            }
+        } catch (Exception $err) {
+            throw ErrorHelper::getSherlError($err, $this->errorFactory->create(EtlErr::EXTRACT_TRANSFORM_LOAD_FAILED));
+        }
+    }
+    /**
+     * Saves a configuration for the ETL process.
+     * @param EtlSaveConfigInputDto $config - The configuration to save.
+     * @return ConfigModelDto|null - The saved configuration model.
+     * @throws SherlException
+     */
+    public function saveConfig(EtlSaveConfigInputDto $config): ?ConfigModelDto
+    {
+        try {
+            $response = $this->client->post("/api/save-config", [
```

Comment:
URL incorrecte, il manque le préfixe de l'ETL : 
```suggestion
            $response = $this->client->post("/api/etl/save-config", [
```